auraed/src/cells/cell_service/cell_service.rs

```diff
@@ -326,4 +366,93 @@ impl cell_service_server::CellService for CellService {
             self.stop_in_cell(&cell_name, request).await
         }
     }
+
+    async fn list(
+        &self,
+        _request: Request<CellServiceListRequest>,
+    ) -> std::result::Result<Response<CellServiceListResponse>, Status> {
+        Ok(Response::new(self.list().await?))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::cells::cell_service::validation::{
+        ValidatedCell, ValidatedCpuController, ValidatedCpusetController,
+    };
+    use iter_tools::Itertools;
+
+    // Ignored: requires sudo, which we don't have in CI
+    #[ignore]
+    #[tokio::test]
+    async fn test_list() {
+        let service = CellService::new();
+
+        let parent_cell_name = format!("ae-test-{}", uuid::Uuid::new_v4());
+        assert!(service
+            .allocate(allocate_request(&parent_cell_name))
+            .await
+            .is_ok());
+
+        let nested_cell_name =
+            format!("{}/ae-test-{}", &parent_cell_name, uuid::Uuid::new_v4());
+        assert!(service
+            .allocate(allocate_request(&nested_cell_name))
+            .await
+            .is_ok());
+
+        let cell_without_children_name =
+            format!("ae-test-{}", uuid::Uuid::new_v4());
+        assert!(service
+            .allocate(allocate_request(&cell_without_children_name))
+            .await
+            .is_ok());
+
+        let result = service.list().await;
+        assert!(result.is_ok());
+
+        let list = result.unwrap();
+        assert_eq!(list.cells.len(), 2);
+
+        let expected_root_cell_names =
+            vec![&parent_cell_name, &nested_cell_name].sort();
```

Comment:
Should this be `cell_without_children_name` instead of `nested_cell_name`?

Reply:
It should! I hadn't realized that `.sort()` doesn't actually return the sorted vector so I was comparing `()` with `()` here. 🤦‍♂️